aws_lambda_powertools/utilities/data_classes/ses_event.py

```diff
@@ -107,18 +131,78 @@ def get_type(self) -> str:
         # Note: this name conflicts with existing python builtins
         return self["type"]
 
+    @property
+    def topic_arn(self) -> Optional[str]:
+        """String that contains the Amazon Resource Name (ARN) of the Amazon SNS topic to which the
+        notification was published."""
+        return self.get("topicArn")
+
     @property
     def function_arn(self) -> str:
         """String that contains the ARN of the Lambda function that was triggered.
+
         Present only for the Lambda action type."""
         return self["functionArn"]
 
     @property
     def invocation_type(self) -> str:
         """String that contains the invocation type of the Lambda function. Possible values are RequestResponse
-        and Event. Present only for the Lambda action type."""
+        and Event.
+
+        Present only for the Lambda action type."""
         return self["invocationType"]
 
+    @property
+    def bucket_name(self) -> str:
+        """String that contains the name of the Amazon S3 bucket to which the message was published.
+
+        Present only for the S3 action type."""
+        return str(self["bucketName"])
```

Comment:
Should we use self.get() on field that may not be present so it returns None instead of a KeyError?

Reply:
In general it is optional, yes. 

But if the action type is "S3", it would be required. Same for all of the other fields for Lambda, Bounce and S3 action types.